src/server/scripts/Shadowlands/Sanctum of Domination/boss_sylvanas_windrunner.cpp

```diff
@@ -0,0 +1,3694 @@
+    }
+};
+
+// Frigid Shards - 354933
+class spell_sylvanas_windrunner_frigid_shards : public AuraScript
+{
+    PrepareAuraScript(spell_sylvanas_windrunner_frigid_shards);
+
+    bool Validate(SpellInfo const* spellInfo) override
+    {
+        return ValidateSpellInfo ({ SPELL_FRIGID_SHARDS_MISSILE });
+    }
+
+    void HandlePeriodic(AuraEffect const* aurEff)
+    {
+        GetCaster()->CastSpell(GetCaster()->GetVictim(), SPELL_FRIGID_SHARDS_MISSILE, true);
+    }
+
+    void Register() override
+    {
+        OnEffectPeriodic += AuraEffectPeriodicFn(spell_sylvanas_windrunner_frigid_shards::HandlePeriodic, EFFECT_0, SPELL_AURA_PERIODIC_DUMMY);
+    }
+};
+
+// Desecrating Shot - 22400
+struct at_sylvanas_windrunner_disecrating_shot : AreaTriggerAI
+{
+    at_sylvanas_windrunner_disecrating_shot(AreaTrigger* areatrigger) : AreaTriggerAI(areatrigger) { }
+
+    void OnCreate() override
+    {
+        _instance = at->GetInstanceScript();
+
+        _updateDiff = 0;
+    }
+
+    void OnUpdate(uint32 diff) override
+    {
+        if (!_instance)
+            return;
+
+        _updateDiff += diff;
+
+        if (_updateDiff >= 3150)
+        {
+            if (Creature* sylvanas = _instance->GetCreature(DATA_SYLVANAS_WINDRUNNER))
+            {
+                if (!sylvanas)
+                    return;
+
+                if (boss_sylvanas_windrunner* ai = CAST_AI(boss_sylvanas_windrunner, sylvanas->AI()))
+                {
+                    if (Creature* shadowCopy = ObjectAccessor::GetCreature(*at, ai->GetShadowCopyJumperGuid()))
+                        shadowCopy->SendPlaySpellVisual(at->GetPosition(), 0, SPELL_VISUAL_DESECRATING_ARROW, 0, 0, 0.280999988317489624f, true);
+
+                    sylvanas->CastSpell(at->GetPosition(), SPELL_DESECRATING_SHOT_TRIGGERED, true);
+
+                    at->Remove();
+                }
+            }
+        }
+    }
+
+private:
+    InstanceScript* _instance;
+    uint32 _updateDiff;
+};
+
+class DebrisEvent : public BasicEvent
+{
+    public:
+        DebrisEvent(Unit* owner, Position const debrisDestPos) : _owner(owner), _debrisDestPos(debrisDestPos) { }
+
+        bool Execute(uint64 /*time*/, uint32 /*diff*/) override
+        {
+            _owner->CastSpell(_debrisDestPos, SPELL_RIVEN_DEBRIS, true);
+            return true;
+        }
+
+    private:
+        Unit* _owner;
+        Position _debrisDestPos;
+};
+
+// Rive - 23028
+struct at_sylvanas_windrunner_rive : AreaTriggerAI
+{
+    at_sylvanas_windrunner_rive(AreaTrigger* areatrigger) : AreaTriggerAI(areatrigger) { }
+
+    void OnCreate() override
+    {
+        _instance = at->GetInstanceScript();  
+    }
+
+    void OnSplineIndexReached(int splineIndex) override
+    {
+        if (!_instance)
+            return;
+
+        switch (splineIndex)
+        {
+            default:
```

Comment:
switch with no cases? xD

Reply:
ehh, yes; I thought using default was correct on this case (?)

Reply:
if you have no case but just default you dont need a switch. To explain it a bit:
```c
switch (something)
{
    case 0:
        // dosomething0
        break;
    case 1:
        // dosomething1
        break;
    default
        // dosomethingelse
        break;
}
```
is equivalent to
```c
if (something == 0)
    // dosomething0
else if (something == 1)
    // dosomething1
else
    // dosomethingelse
```
with ifs you couldnt even do that magic which you did. What you literally used (if we convert it to ifs) is simply using an else block without and if block xD

Reply:
that makes sense, I'll remove the switch.

fixed on next commit ☄️